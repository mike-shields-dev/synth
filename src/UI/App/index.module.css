.App {
    display: grid;
    grid-template-columns: 1fr;
    grid-template-rows: 2rem minmax(0, 1fr) 4rem;
    grid-template-areas: 
        'header'
        'main'
        'footer'
    ;
    height: calc(100vh - 2rem);
    gap: 0.5rem;
}

.App__Header {
    grid-area: header;
}

.App__Main {
    grid-area: main;
    overflow-y: auto;
    overflow-x: hidden;
    scroll-behavior: smooth;
}

.App__Footer {
    grid-area: footer;
}
