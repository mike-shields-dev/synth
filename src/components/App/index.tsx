import React from 'react';
import { useSessionStorage } from '../../hooks/useSessionStorage';
import { Synth } from '../Synth';
import { SynthParameterGroup } from '../SynthParameterGroup';
import './index.module.css';

function App() {
  const [focus, setFocus] = useSessionStorage('focus', "");

  function updateFocus(id: string) {
    setFocus(id);
  } 

  return (
    <div className="App">
      <Synth>
        <SynthParameterGroup 
          groupName="filter" 
          updateFocus={updateFocus}
          isFocused={focus === "filter"}
        >
          <input type="range" />
        </SynthParameterGroup>

        <SynthParameterGroup 
          groupName="filterEnvelope"
          updateFocus={updateFocus}
          isFocused={focus === "filterEnvelope"}
        >
          <input type="range" />
        </SynthParameterGroup>
      </Synth> 
    </div>
  )
}

export default App
