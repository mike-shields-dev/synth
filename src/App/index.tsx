import { useState } from 'react';
import { AmpEnvelope } from '../components/AmpEnv';
import { Filter } from '../components/Filter';
import { FilterEnvelope } from '../components/FilterEnvelope';
import { Oscillator } from '../components/Oscillator';
import { SynthUI } from '../components/SynthUI';
import { useMidiAccess } from '../hooks/useMidiAccess';
import './index.module.css';

function App() {
  const [focus, setFocus] = useState('oscillator');
  useMidiAccess();

  function updateFocus(id: string) {
    setFocus(id);
  }

  return (
    <>
    <div className="App">
      <SynthUI>
        <Oscillator {...{ focus, updateFocus }} />
        <Filter {...{ focus, updateFocus }} />
        <FilterEnvelope {...{ focus, updateFocus }} />
        <AmpEnvelope {...{ focus, updateFocus }} />
      </SynthUI>
    </div>
    </>
  )
}

export default App


