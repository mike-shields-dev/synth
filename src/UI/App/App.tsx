import { useState } from 'react';
import Nav from '../components/Nav';
import { AmpEnvelope, FilterEnvelope, Oscillator, SynthUI } from '../components';
import { Filter } from '../components/Filter';
import { useMidiAccess } from '../hooks/useMidiAccess';
import './index.module.css';
import { publishFocusChange } from '../../PubSub/FocusChange';
import { UIKeyboard } from '../components/UIKeyboard';
import css from './App.module.css';

function App() {
  const [focus, setFocus] = useState('oscillator');
  useMidiAccess();

  function updateFocus(focus: string) {
    setFocus(focus);
    publishFocusChange(focus);
  }

  return (
    <>
      <div className={css.App}>

        <div className={css.App__Header}>
          <Nav {...{ focus, updateFocus }} />
        </div>

        <div className={css.App__Main}>
          <SynthUI>
            <Oscillator {...{ focus, updateFocus }} />
            <Filter {...{ focus, updateFocus }} />
            <FilterEnvelope {...{ focus, updateFocus }} />
            <AmpEnvelope {...{ focus, updateFocus }} />
          </SynthUI>
        </div>

        <div className={css.App__Footer}>
          <UIKeyboard />
        </div>
      </div>
    </>
  )
}

export default App
