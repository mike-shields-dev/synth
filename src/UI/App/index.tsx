import { useState } from 'react';
import { AmpEnvelope, FilterEnvelope, Oscillator, SynthUI } from '../components';
import { Filter } from '../components/Filter';
import { useMidiAccess } from '../hooks/useMidiAccess';
import './index.module.css';
import * as Tone from 'tone';
import { synth } from '../../Synth';


function App() {
  const [focus, setFocus] = useState('oscillator');
  useMidiAccess();

  function updateFocus(id: string) {
    setFocus(id);
    synth.focus = id;
  }

  return (
    <>
    <div className="App">
        <button onClick={async () => await Tone.start()}>Audio</button>
        <SynthUI>
        <Oscillator {...{ focus, updateFocus }} />
        <Filter {...{ focus, updateFocus }} />
        <FilterEnvelope {...{ focus, updateFocus }} />
        <AmpEnvelope {...{ focus, updateFocus }} />
      </SynthUI>
    </div>
    </>
  )
}

export default App
