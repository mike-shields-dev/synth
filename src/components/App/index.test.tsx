import { render, screen } from '@testing-library/react';
import App from './index';

describe('App', () => {
    beforeEach(() => { render(<App />) });

    it('matches snapshot', () => {
        const { container } = render(<App />);

        expect(container.firstChild).toMatchSnapshot();
    });

    it('renders the SynthUI', () => {
        expect(screen.getByTestId(/SynthUI/i)).toBeInTheDocument();
    });

    it('renders a SynthParameterGroup for the oscillator', () => {
        expect(screen.getByRole('form', { name: 'oscillator' })).toBeInTheDocument();
    });

    it('renders a SynthParameterGroup for the filter', () => {
        expect(screen.getByRole('form', { name: 'filter' })).toBeInTheDocument();
    });

    it('renders a SynthParameterGroup for the filter', () => {
        expect(screen.getByRole('form', { name: 'filterEnvelope' })).toBeInTheDocument();
    });

    it('renders a SynthParameterGroup for the filter', () => {
        expect(screen.getByRole('form', { name: 'envelope' })).toBeInTheDocument();
    });

    it('renders a SynthParameterGroup form the (amp) envelope', () => {
        expect(screen.getByRole('form', { name: 'envelope' })).toBeInTheDocument();
    })
});