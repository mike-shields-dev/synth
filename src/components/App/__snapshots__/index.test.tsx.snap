// Vitest Snapshot v1

exports[`App > matches snapshot 1`] = `
<div
  class="App"
>
  <div
    data-testid="SynthUI"
  >
    <section>
      <form
        aria-label="oscillator"
        class="_SynthParameterGroup_f01a7f"
        id="oscillator"
      >
        <header>
          <h2>
            Oscillator
          </h2>
        </header>
        <input
          type="range"
        />
      </form>
    </section>
    <section>
      <form
        aria-label="filter"
        class="_SynthParameterGroup_f01a7f"
        id="filter"
      >
        <header>
          <h2>
            Filter
          </h2>
        </header>
        <input
          type="range"
        />
      </form>
    </section>
    <section>
      <form
        aria-label="filterEnvelope"
        class="_SynthParameterGroup_f01a7f"
        id="filterEnvelope"
      >
        <header>
          <h2>
            Filter Envelope
          </h2>
        </header>
        <input
          type="range"
        />
      </form>
    </section>
    <section>
      <form
        aria-label="envelope"
        class="_SynthParameterGroup_f01a7f"
        id="envelope"
      >
        <header>
          <h2>
            Amp Envelope
          </h2>
        </header>
        <input
          type="range"
        />
      </form>
    </section>
  </div>
</div>
`;
